lib/openstack_query/api/query_api.py

```diff
@@ -285,3 +285,28 @@ def append_from(
 
         self.output.update_forwarded_outputs(link_props[0], new_query.to_props())
         return self
+
+    def reset(self, hard=False):
```

Comment:
Bike shedding, reset should reset everything

Reply:
disagree - we can use reset() for this use-case:

you run a query and group-by/sort-by 

you want to output same query results but without grouping and sorting - `reset(hard=False)` means they don't have to run the query again